bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details2/NonSiderealNameEditor.scala

```diff
@@ -0,0 +1,152 @@
+package jsky.app.ot.gemini.editor.targetComponent.details2
+
+import java.awt.Color
+import java.util.Date
+import javax.swing.{SwingUtilities, JOptionPane, JLabel}
+
+import edu.gemini.horizons.server.backend.HorizonsService2
+import edu.gemini.horizons.server.backend.HorizonsService2.HS2
+import edu.gemini.pot.sp.ISPNode
+import edu.gemini.shared.gui.GlassLabel
+import edu.gemini.shared.util.immutable.{Option => GOption}
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.spModel.core.{Site, Ephemeris, HorizonsDesignation, Target}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.system.ConicTarget
+
+import jsky.app.ot.gemini.editor.targetComponent.TelescopePosEditor
+import jsky.util.gui.{DialogUtil, TextBoxWidget, TextBoxWidgetWatcher}
+
+
+import Horizons.HorizonsIO
+
+import scala.swing.Swing
+import scalaz._, Scalaz._, effect.IO
+
+final class NonSiderealNameEditor extends TelescopePosEditor with ReentrancyHack {
+
+  private[this] var site = Option.empty[Site]
+  private[this] var spt = new SPTarget // never null
+
+  def lookup(site: Option[Site]): Unit = {
+    import HorizonsService2.{ Search, Row }, Search._
+
+    /// Some IO actions for looking up targets and epherimides
+
+    val askSearch: HS2[Option[Search[_ <: HorizonsDesignation]]] =
+      HS2.delay {
+        case class Wrapper(unwrap: Search[_ <: HorizonsDesignation]) {
+          override def toString = unwrap.productPrefix
+        }
+        Option {
+          JOptionPane.showInputDialog(
+            name,
+            "Select the HORIZONS target type:",
+            "Horizons Search",
+            JOptionPane.QUESTION_MESSAGE,
+            null, // TODO: icon
+            List(Comet, Asteroid, MajorBody).map(f => Wrapper(f(name.getText))).toArray[Object],
+            null)
+        } .map(_.asInstanceOf[Wrapper].unwrap)
+      }
+
+    def show(msg: String): HS2[Unit] =
+      HS2.delay(Swing.onEDT(GlassLabel.show(SwingUtilities.getRootPane(name), msg)))
+
+    def hide: HS2[Unit] =
+      HS2.delay(Swing.onEDT(GlassLabel.hide(SwingUtilities.getRootPane(name))))
+
+    def lookup(d: HorizonsDesignation, site: Site): HS2[Ephemeris] =
+      show("Fetching Ephemeris...") *> HorizonsService2.lookupEphemeris(d, site, 1000) // arbitrary
```

Comment:
I guess here we are showing the glass pane while doing the lookup, nice code :+1: